### PR TITLE
Local Sounds

### PR DESCRIPTION
**Pull request by MarinaGryphon**
- tweak: "Sounds should now no longer be able to be heard across the station."
  - tweak: "Messes with the volume of some sounds to make them less obnoxiously loud or far too quiet."

### CODE REVIEW

**Comment by Roland410** [comment]
I know this should be in suggestions (and it was) but a jukebox toggle maybe? Some people find it annoying to listen to shitty music someone set.

File: code/game/sound.dm [hunk 1]
```diff
@@ -54,112 +54,158 @@ GLOBAL_LIST_INIT(button_sound,list('sound/machines/button1.ogg','sound/machines/
...
-			continue
-		if(get_dist(M, turf_source) <= (world.view + extrarange) * 2)
-			var/turf/T = get_turf(M)
-			if(T && T.z == turf_source.z && (!is_ambiance || M.get_preference_value(/datum/client_preference/play_ambiance) == GLOB.PREF_YES))
-				M.playsound_local(turf_source, soundin, vol, vary, frequency, falloff, is_global, extrarange)
+	if (is_global)
+		playsound_allinrange(source, original_sound,
+			extra_range = extrarange,
+			is_global = is_global,
+			use_random_freq = !!vary,
```
Comment (BurgerLUA) [inline code/game/sound.dm]: What does the double exclamation mark mean in code? Pretty sure that means NOT NOT.

File: code/game/sound.dm [hunk 2]
```diff
@@ -54,112 +54,158 @@ GLOBAL_LIST_INIT(button_sound,list('sound/machines/button1.ogg','sound/machines/
...
+			use_pressure = usepressure,
+			modify_environment = (environment != 0),
+			is_ambiance = is_ambiance
+		)
+	else
+		playsound_lineofsight(source, original_sound,
+			use_pressure = usepressure,
+			use_random_freq = !!vary,
```
Comment (BurgerLUA) [inline code/game/sound.dm]: Same thing here with the double exclamation mark.

File: code/game/sound.dm (hunk 1 above)
Comment (MarinaGryphon) [reply code/game/sound.dm]: Yes. It converts it to a boolean.

**Pull request merged**